src/openms/source/METADATA/SpectrumMetaDataLookup.cpp

```diff
@@ -215,4 +213,51 @@ namespace OpenMS
     return success;
   }
 
+  bool SpectrumMetaDataLookup::addMissingSpectrumReferences(vector<PeptideIdentification>& peptides, const String& filename,
+    bool stop_on_error, bool override_spectra_data, vector<ProteinIdentification> proteins)
+  {
+    bool success = true;
+    PeakMap exp;
+    SpectrumMetaDataLookup lookup;
+    if (lookup.empty())
+    {
+      FileHandler().loadExperiment(filename, exp);
+      lookup.readSpectra(exp.getSpectra());
+      lookup.setSpectraDataRef(filename);
+    }
+    if (override_spectra_data)
+    {
+      vector<String> spectra_data(1);
+      spectra_data[0] = "file://" + lookup.spectra_data_ref;
+      for (vector<ProteinIdentification>::iterator it =
+            proteins.begin(); it !=
+            proteins.end(); ++it)
+      {
+        it->setMetaValue("spectra_data", spectra_data);
+      }
+    }
+    for (vector<PeptideIdentification>::iterator it =
+          peptides.begin(); it !=
+          peptides.end(); ++it)
+    {
+      try
+      {
+        Size index = lookup.findByRT(it->getRT());
+        SpectrumMetaDataLookup::SpectrumMetaData meta;
+        lookup.getSpectrumMetaData(index, meta);
+        it->setMetaValue("spectrum_reference", meta.native_id);
+      }
+      catch (Exception::ElementNotFound&)
+      {
+        LOG_ERROR << "Error: Failed to look up spectrum native ID for peptide ID with retention time '" + String(it->getRT()) + "'." << endl;
```

Comment:
I think I took that over from SpectrumMetaDataLookup::addMissingSpectrumReferences. Clarified in both places.